include/caffe/util/io.hpp

```diff
@@ -9,26 +11,43 @@
 #include "caffe/common.hpp"
 #include "caffe/proto/caffe.pb.h"
 
+#ifndef CAFFE_TMP_DIR_RETRIES
+#define CAFFE_TMP_DIR_RETRIES 100
+#endif
+
 namespace caffe {
 
 using ::google::protobuf::Message;
 using ::boost::filesystem::path;
 
-inline void MakeTempFilename(string* temp_filename) {
-  temp_filename->clear();
-  const path& model = boost::filesystem::temp_directory_path()
-    /"caffe_test.%%%%%%";
-  *temp_filename = boost::filesystem::unique_path(model).string();
-}
+static path temp_files_subpath;
+static uint64_t next_temp_file = 0;
```

Comment:
Better to move this two static variables into function `MakeTempFilename` so that they are only accessible in that function.